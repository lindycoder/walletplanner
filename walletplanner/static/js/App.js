import React from 'react';
import AddTransactions from './AddTransactions';
import TransactionsView from './TransactionsView';
import {StaticMonthly} from './Periodicity';
import {moment} from './utils';
import PropTypes from 'prop-types';

class App extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            transactions: []
        }
    }

    render() {
        return (
            <div className="App container">
                <h1 className="display-4">Wallet Planner</h1>
                <AddTransactions server={this.props.server}
                                 onTransactionAdded={this.refreshTransactions.bind(this)}/>
                <TransactionsView transactions={this.state.transactions}
                                  periodicity={new StaticMonthly([1, 15])}
                                  currentDate={moment().startOf('day')}
                                  periodStart={-3}
                                  openingBalance={0}/>
            </div>
        )
    }

    refreshTransactions() {
        this.props.server.getTransactions()
            .then(ts => this.setState({transactions: ts}));
    }
}

App.propTypes = {
    homeUrl: PropTypes.string.isRequired,
    echoUrl: PropTypes.string.isRequired,
    server: PropTypes.object.isRequired,
};

export default App;
