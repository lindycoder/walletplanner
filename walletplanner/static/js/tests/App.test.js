import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('App', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        Date.now = jest.fn(() => new Date(2019, 1, 2));

        props = {
          homeUrl: '/home-url',
          echoUrl: '/echo-url',
          server: {getTransactions: jest.fn()},
        };
        wrapper = shallow(<App {...props} />);
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('On transaction added, should refresh the transaction list', async () => {
        const transactionList = [{amount: 100}, {amount: 200}];

        props.server.getTransactions.mockResolvedValue(transactionList);

        wrapper.find('AddTransactions').simulate('transactionAdded');
        await undefined;

        expect(wrapper.state().transactions).toBe(transactionList);

        expect(wrapper).toMatchSnapshot();
    });
});
