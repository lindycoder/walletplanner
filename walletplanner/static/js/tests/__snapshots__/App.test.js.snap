// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App On transaction added, should refresh the transaction list 1`] = `
<div
  className="App container"
>
  <h1
    className="display-4"
  >
    Wallet Planner
  </h1>
  <AddTransactions
    onTransactionAdded={[Function]}
    server={
      Object {
        "getTransactions": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": Promise {},
            },
          ],
        },
      }
    }
  />
  <TransactionsView
    currentDate={"2019-02-02T00:00:00.000Z"}
    periodStart={-3}
    periodicity={
      StaticMonthly {
        "days": Array [
          1,
          15,
        ],
      }
    }
    transactions={
      Array [
        Object {
          "amount": 100,
        },
        Object {
          "amount": 200,
        },
      ]
    }
  />
</div>
`;

exports[`App should render correctly 1`] = `
<div
  className="App container"
>
  <h1
    className="display-4"
  >
    Wallet Planner
  </h1>
  <AddTransactions
    onTransactionAdded={[Function]}
    server={
      Object {
        "getTransactions": [MockFunction],
      }
    }
  />
  <TransactionsView
    currentDate={"2019-02-02T00:00:00.000Z"}
    periodStart={-3}
    periodicity={
      StaticMonthly {
        "days": Array [
          1,
          15,
        ],
      }
    }
    transactions={Array []}
  />
</div>
`;
